src/Commands/Workflow/WaitForCommitCommand.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Workflow;
+
+use Pantheon\Terminus\Collections\WorkflowLogsCollection;
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Pantheon\Terminus\Exceptions\TerminusException;
+use Pantheon\Terminus\Models\WorkflowLog;
+
+class WaitForCommitCommand extends TerminusCommand
+{
+    /**
+     * Wait for a workflow to complete. Usually this will be used to wait
+     * for code commits, since Terminus will already wait for workflows
+     * that it starts through the API.
+     *
+     * @command workflow:wait:commit
+     * @param string $site_id The pantheon site to wait for.
+     * @param string $target_commit commit id to wait for
+     * @option max Maximum number of seconds to wait for the workflow to complete (default 600)
+     */
+    public function waitForCommit(
+        string $site_id,
+        string $target_commit,
+        $options = ['max' => 600]
+    ) {
+        $wfl = null;
+        $site = $this->getSiteById($site_id);
+        $wflc = $site->getWorkflowLogs();
+        if (!$wflc instanceof WorkflowLogsCollection) {
+            throw new TerminusException('Workflow logs could not be retrieved for site: {site}', ['site' => $site_id,]);
+        }
+
+        // Find the latest workflow that matches the commit hash
+        $wfl = $wflc->findLatestFromOptionsArray([
```

Comment:
It seems that we only use this function with `target_commit`, so we could probably simplify it to only work for that one property. We don't need to do that now -- or ever, if you expect that future expansion of the functionality might exercise more of this functionality.